### PR TITLE
Warnint util 4516 v2

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:
https://redmine.openinfosecfoundation.org/issues/4516

Describe changes:
- Fix integer warnings `-Wimplicit-int-conversion` in all util files

Modifies #6791 
- also removes warnings from unit tests
- fixes radix tree loop (using uint, we can no longer have condition >= 0)

### CODE REVIEW

File: src/util-decode-mime.c [hunk 1]
```diff
@@ -687,7 +687,11 @@ static uint8_t * GetFullValue(DataValue *dv, uint32_t *len)
 static inline uint8_t * FindBuffer(const uint8_t *src, uint32_t len, const uint8_t *find, uint32_t find_len)
 {
     /* Use utility search function */
-    return BasicSearchNocase(src, len, find, find_len);
+    if (find_len > UINT16_MAX) {
+        // TODO rewrite all this to have app-layer events
+        find_len = UINT16_MAX;
+    }
+    return BasicSearchNocase(src, len, find, (uint16_t)find_len);
```
Comment (victorjulien) [inline src/util-decode-mime.c:694]: I wonder if we should just update `BasicSearchNocase` to accept `uint32_t`?

Comment (catenacyber) [reply src/util-decode-mime.c:694]: `len` is `uint32_t`(haystack)
The needle is `uint16_t`
That looks reasonable...

**Comment by suricata-qa** [comment]
Information:
 
 | field | test | baseline | % |
|--|--|--|--| 
| | tlpr1_stats_chk |
| .flow.memuse | 524447808 | 553066368 | 94.83% | 

Pipeline 5699

File: src/util-decode-mime.c (hunk 1 above)
Comment (victorjulien) [reply src/util-decode-mime.c:694]: ok, makes sense. Can you add a ticket for tracking the events setting?

**Comment by victorjulien** [comment]
replaced by #6820

File: src/util-decode-mime.c (hunk 1 above)
Comment (catenacyber) [reply src/util-decode-mime.c:694]: Or should this mime code get rustified first ?

Comment (victorjulien) [reply src/util-decode-mime.c:694]: For master that would be nice. Would we need a backport for this int handling though?

Comment (catenacyber) [reply src/util-decode-mime.c:694]: I do not know.
It will be a big work I guess...
I am not sure it is worth it...

**Pull request closed without merging**